activesupport/lib/active_support/notifications/fanout.rb

```diff
@@ -39,8 +40,10 @@ def unsubscribe(subscriber_or_name)
           when String
             @string_subscribers[subscriber_or_name].clear
             @listeners_for.delete(subscriber_or_name)
+            @other_subscribers.each { |sub| sub.blacklist!(subscriber_or_name) }
```

Comment:
Can you change `blacklist!` to "restricted_list" or maybe `restrict!`.

Reply:
How about `unsubscribed`?